src/current/v23.2/changefeed-messages.md

```diff
@@ -191,70 +191,134 @@ CREATE CHANGEFEED FOR TABLE rides INTO 'external://kafka' WITH diff, envelope=wr
+1. Insert and update values in `employees`:
 
-    ~~~json
-    [1]	{"__crdb__": {"updated": <timestamp 1>}, "id": 1, "name": "Carl"}
-    [1]	{"__crdb__": {"updated": <timestamp 2>}, "id": 1, "name": "Petee"}
-    [1]	{"__crdb__": {"updated": <timestamp 1>}, "id": 1, "name": "Carl"}
+    {% include_cached copy-clipboard.html %}
+    ~~~ sql
+    INSERT INTO employees VALUES (1, 'Terry', 'new york city');
+    INSERT INTO employees VALUES (2, 'Alex', 'los angeles');
+    UPDATE employees SET name = 'Terri' WHERE id = 1;
+    INSERT INTO employees VALUES (3, 'Ash', 'london');
+    UPDATE employees SET name = 'Terrence' WHERE id = 1;
+    UPDATE employees SET office = 'new york city' WHERE id = 2;
+    INSERT INTO employees VALUES (4, 'Danny', 'los angeles');
+    INSERT INTO employees VALUES (5, 'Robbie', 'london');
     ~~~
 
-    However, you will **never** see an output like the following (i.e., an out of order row that you've never seen before):
+    {{site.data.alerts.callout_info}}
+    In a [transaction]({% link {{ page.version.version }}/transactions.md %}), if a row is modified more than once in the same transaction, the changefeed will only emit the last change.
+    {{site.data.alerts.end}}
 
-    ~~~json
-    [1]	{"__crdb__": {"updated": <timestamp 2>}, "id": 1, "name": "Petee"}
-    [1]	{"__crdb__": {"updated": <timestamp 1>}, "id": 1, "name": "Carl"}
+1. The sink will receive messages of the inserted rows emitted per timestamp:
+
+    ~~~
+    {"after": {"id": 1, "name": "Terry", "office": "new york city"}, "key": [1], "updated": "1701102296662969433.0000000000"}
+    {"after": {"id": 1, "name": "Terri", "office": "new york city"}, "key": [1], "updated": "1701102311425045162.0000000000"}
+    {"after": {"id": 2, "name": "Alex", "office": "los angeles"}, "key": [2], "updated": "1701102305519323705.0000000000"}
+    {"after": {"id": 3, "name": "Ash", "office": "london"}, "key": [3], "updated": "1701102316388801052.0000000000"}
+    {"after": {"id": 1, "name": "Terrence", "office": "new york city"}, "key": [1], "updated": "1701102320607990564.0000000000"}
+    {"after": {"id": 2, "name": "Alex", "office": "new york city"}, "key": [2], "updated": "1701102325724272373.0000000000"}
+    {"after": {"id": 5, "name": "Robbie", "office": "london"}, "key": [5], "updated": "1701102330377135318.0000000000"}
+    {"after": {"id": 4, "name": "Danny", "office": "los angeles"}, "key": [4], "updated": "1701102561022789676.0000000000"}
     ~~~
 
-- If a row is modified more than once in the same transaction, only the last change will be emitted.
+    The messages received at the sink are in order by timestamp **for each key**. Here, the update for key `[1]` has emitted before the insertion of key `[2]` even though the timestamp for the update to key `[1]` is higher. If you order messages by key, they will always be in the correct timestamp order.
```

Comment:
> If you order messages by key, they will always be in the correct timestamp order.

This reads a little confusing to me. We're talking about the ordering in which the sink sees updates _for each key_?

Could this say something like "Put differently, if you follow the sequence of updates _for a given key_ at the sink, they will always be in the correct timestamp order."?

Reply:
Yes, I have adjusted this sentence for your suggestion here. Thanks!